src/main/java/org/folio/bulkops/service/BulkOperationService.java

```diff
@@ -301,26 +312,35 @@ public void confirm(BulkOperation operation)  {
 
       operation.setLinkToModifiedRecordsJsonFile(modifiedJsonFileName);
 
-      dataProcessing.setProcessedNumOfRecords(processedNumOfRecords);
-      dataProcessingRepository.save(dataProcessing);
+      updateProcessed(dataProcessing, operation, processedNumOfRecords);
 
-      operation.setApproach(IN_APP);
-      operation.setStatus(OperationStatusType.REVIEW_CHANGES);
-      operation.setProcessedNumOfRecords(processedNumOfRecords);
       bulkOperationRepository.findById(operation.getId()).ifPresent(op -> operation.setCommittedNumOfErrors(op.getCommittedNumOfErrors()));
+    } catch (S3ClientException e) {
+      errorMsg = ERROR_NOT_CONFIRM_CHANGES_S3_ISSUE;
     } catch (Exception e) {
-      log.error(e);
-      dataProcessingRepository.save(dataProcessing
-        .withStatus(StatusType.FAILED)
-        .withEndTime(LocalDateTime.now()));
-      operation.setStatus(OperationStatusType.FAILED);
-      operation.setEndTime(LocalDateTime.now());
-      operation.setErrorMessage(ERROR_NOT_CONFIRM_CHANGES_S3_ISSUE);
+      errorMsg = e.getMessage();
     } finally {
+      if (nonNull(errorMsg)) {
+        dataProcessingRepository.save(dataProcessing
```

Comment:
Using error message as flag makes code dirty. I would propose extract this logic in separate method and use it with in each catch-block with different error messages.